packages/office-ui-fabric-react/src/utilities/decorators/withViewport.tsx

```diff
@@ -44,11 +46,38 @@ export function withViewport<P extends { viewport?: IViewport }, S>(ComposedComp
         });
 
       this._events.on(window, 'resize', this._onAsyncResize);
+
+      this._observer = new MutationObserver((mutations) => {
+        mutations.forEach((mutation) => {
+        if (mutation.type === 'attributes') {
+          if (mutation.attributeName === 'style') {
+            if ((mutation.oldValue.indexOf('display:none') !== -1) || (mutation.oldValue.indexOf('visibility:hidden') !== -1)) {
+              this._updateViewport(true);
+              return;
+            }
+          }
+          if (mutation.attributeName === 'class') {
+            if (mutation.oldValue.indexOf('hide') !== -1) {
```

Comment:
It seems very unpredictable, in that it is unclear how to know for sure how the user is showing or hiding the viewport. Jon had mentioned trying to detect overall page size. I saw some notes around sizechanged that might be worth investigation. Looking into that now.